### PR TITLE
improve(project): fix context passing and improve project APIs

### PR DESCRIPTION
**Pull request by saitofun**
iss #440

### CODE REVIEW

File: cmd/srv-applet-mgr/global/config.go [hunk 1]
```diff
@@ -30,6 +30,7 @@ import (
 var (
 	App         *confapp.Ctx
 	WithContext contextx.WithContext
```
Comment (saitofun) [inline cmd/srv-applet-mgr/global/config.go:32]: hidden this?

File: pkg/modules/project/project.go [hunk 2]
```diff
@@ -4,519 +4,203 @@ package project
...
+	return m, nil
 }
 
-func (r *ListProjectReq) Additions() builder.Additions {
-	m := &models.Project{}
-	return builder.Additions{
-		builder.OrderBy(builder.DescOrder(m.ColCreatedAt())),
-		r.Pager.Addition(),
+func GetDetail(ctx context.Context, prj *models.Project) (*Detail, error) {
+	rsp, err := applet.ListDetail(ctx, prj.ProjectID, &applet.ListReq{})
```
Comment (saitofun) [inline pkg/modules/project/project.go:53]: applet.ListDetail, set projectID arg to applet.ListReq

File: pkg/modules/project/project.go [hunk 3]
```diff
@@ -4,519 +4,203 @@ package project
...
 			return nil
 		},
+	).Do()
+	if err != nil {
+		return nil, err
+	}
+
+	if err = mq.CreateChannel(ctx, prj.Name, event.Handler); err != nil {
+		conflog.FromContext(ctx).WithValues("prj", prj.Name).
+			Warn(errors.New("channel create failed"))
```
Comment (saitofun) [inline pkg/modules/project/project.go:154]: log error

**Review by saitofun — COMMENTED** [review]
self review

**Review by huangzhiran — APPROVED** [review]

**Review by hunshenshi — APPROVED** [review]

**Pull request merged**